src/wallet/wallet.cpp

```diff
@@ -3465,11 +3465,19 @@ std::unique_ptr<SigningProvider> CWallet::GetSolvingProvider(const CScript& scri
 
 std::unique_ptr<SigningProvider> CWallet::GetSolvingProvider(const CScript& script, SignatureData& sigdata) const
 {
-    for (const auto& spk_man_pair : m_spk_managers) {
-        if (spk_man_pair.second->CanProvide(script, sigdata)) {
-            return spk_man_pair.second->GetSolvingProvider(script);
+    // Search the cache first
+    const auto& it = m_cached_spks.find(script);
+    if (it != m_cached_spks.end()) {
+        for (const auto& spkm : it->second) {
+            if (spkm->CanProvide(script, sigdata)) {
```

Comment:
What is the difference between this (a966473) and the previous commit (f0c844bda) in terms of `CanProvide` usage?

In the previous commit, `CanProvide` usage was removed, while here you kept it.
I understand that you removed the `CanProvide` call from the previous commit because it is redundant (it calls IsMine internally which is exactly what `m_cached_spks` does) but I also agree that we might want to keep `CanProvide` because its internal implementation could vary over time.

Reply:
I think this may have just been a holdover from when I was investigating using the same cache for signing. There should be no difference as `CanProvide` is redundant in both contexts.

Reply:
I've dropped this `CanProvide`.